spec/controllers/discovery_controller_spec.rb

```diff
@@ -19,7 +19,7 @@
 
         'scopes_supported' => ['openid'],
 
-        'response_types_supported' => ['code'],
+        'response_types_supported' => ['code', 'token', 'id_token', 'id_token token'],
```

Comment:
Is it OK that before response types supported didn't have this values? Or it's my stupid mistake and I need to register this custom grant flows not as authorization, but as token grants?

Reply:
@nbulaj good catch! :thinking: 

I think they should only be included if implicit flows are enabled, which doesn't seem to be the case in either the default settings or our spec dummy app. (and we should add specs for this case :grin:)

So previously we were including them conditionally in https://github.com/doorkeeper-gem/doorkeeper-openid_connect/pull/116/files#diff-0c58b0448c601da9348f2f321985fe19, but now with the new `Doorkeeper::GrantFlow` they're always included?

This definitely smells like a bug somewhere. Maybe the problem is that we're unconditionally adding the aliased grants at https://github.com/doorkeeper-gem/doorkeeper/blob/b71c5afbfb6694cc023e30be84b9018b5280bf0e/lib/doorkeeper/config.rb#L600?

Reply:
Whithout aliases Doorkeeper would include all the OID grants in the list of enabled... Yep, there definitely could be some issue and I need to find it before the release :hand_over_mouth:

Reply:
> So previously we were including them conditionally in https://github.com/doorkeeper-gem/doorkeeper-openid_connect/pull/116/files#diff-0c58b0448c601da9348f2f321985fe19, but now with the new Doorkeeper::GrantFlow they're always included?

Previously it was like this:

https://github.com/doorkeeper-gem/doorkeeper-openid_connect/blob/master/lib/doorkeeper/openid_connect/response_types_config.rb#L6-L18

So if user configured Doorkeeper as:

```ruby
Doorkeeper.configure do

  grant_flows ["implicit_oidc"]
end
```

It converted to three more grant flows and [resolved by `Doorkeeper::Request` class](https://github.com/doorkeeper-gem/doorkeeper/blob/8fec40d5b873dc62e8bc29b8ebfe71a003b09dab/lib/doorkeeper/request.rb#L30-L33).

But now with `GrantFlow` API all the custom grant flows (like `id_token`) must be explicitly configured and registered, so Doorkeeper could understand:

* what is `implicit_oidc` grant flow
* and how to process it

Since `implicit_oidc` is actually an alias (or how to name it?) to three other grants - we must consider this case.

Reply:
Oh, actually this is the issue!

> Maybe the problem is that we're unconditionally adding the aliased grants at

I'll fix it, thannx for fresh eyes :D :+1:

Reply:
OK, I fixed Doorkeeper itself and defined grant flows in dummy app for specs to test